### ISSUE TITLE
Schema spec: Placeholder discussion issue for indexes on relation fields

### ISSUE THREAD

**Issue by tomhoule** [prisma/specs#280]

**Comment by tomhoule** [comment]
Comparison point: `table.references` in ActiveRecord migrations doesn't create an index by default, apparently ([source](https://apidock.com/rails/ActiveRecord/ConnectionAdapters/Table/references)).

**Comment by mavilein** [comment]
<img width="385" alt="image" src="https://user-images.githubusercontent.com/2470881/67757876-e7799f00-fa3c-11e9-9282-e6fb559c9b04.png">

**Comment by mavilein** [comment]
by @sorenbs :
<img width="368" alt="image" src="https://user-images.githubusercontent.com/2470881/67757892-f2343400-fa3c-11e9-9f69-e6b948567659.png">

**Comment by mavilein** [comment]
I think in 99% you want an index on those relation fields. You would only not want it if the given table is very write heavy and the reads are so rare that you don't need the index. That sounds fairly arcane 😛 
Hence the default should be that there is an index but you can opt out of that behavior. However the rule might need to be a little more sophisticated than using an index all the time. Consider this example for an explicit model for a join table:
```
model UserToPosts {
  post Post
  user User
  @@unique([user, post])
}
```

In this case this simple rule would lead to 3 indexes being created:
* an index on the `post` column
* an index on the `user` column
* a unique constraint index on `(user, post)`

The second index is unnecessary as it can be covered by the third one.

**Comment by pimeys** [comment]
In cases where you need to write a lot, which is basically advertising/marketing industry, you write to a relational database and reflect the data to a column database for analysis anyhow.

**Comment by martonlanga** [comment]
Is there a way to add them manually right now?
Found this in the docs: `@@pg.index([ email, first_name ], name: "my_index", partial: true)` but if I try to define this I get: `Attribute not known: "@pg.index".`

Querying with multiple relations takes around ~8s for me, and sometimes this causes my lambda function to time out, so I'm really looking forward to this!

**Comment by matthewmueller** [comment]
I think we're going to have a lot of these cases, where the 90% case is wanting to take advantage of what the database offers. The 10% case, where you have write-heavy apps or performance-critical apps, where you want to add constraints at the application layer, we need to support that too.

The question to me is in this 10% case, does the schema even need to care about these cases? It's not clear to me what guarantees we can on our end (automatically) that will be more performant than what an index offers. My hunch is that when you opt-out of using indexes, the application constraint logic will be application specific.

In other words, is this good enough for the opt-out case?

```diff
model UserToPosts {
  post Post
  user User
- @@unique([user, post])
}
```
